src/main/java/com/emc/ecs/sync/storage/azure/AzureBlobStorage.java

```diff
@@ -0,0 +1,341 @@
+                    for (ListBlobItem blobItem : ((CloudBlobDirectory) blob).listBlobs(prefix, true, listingDetails, options, opContext)) {
+                        listAllBlobSummaryWithSnapshots(originalBlobMap, blobItem, prefix, listingDetails, options, opContext);
+                    }
+                } catch (URISyntaxException | StorageException e) {
+                    throw new RuntimeException("error for list all Azure blobs");
+                }
+            } else if (blob instanceof CloudBlob) {
+                CloudBlob cloudBlob = (CloudBlob) blob;
+                if (originalBlobMap.containsKey(cloudBlob.getName())) {
+                    originalBlobMap.get(cloudBlob.getName()).add(cloudBlob);
+                } else {
+                    List<CloudBlob> cloudBlobList = new ArrayList<>();
+                    cloudBlobList.add(cloudBlob);
+                    originalBlobMap.put(cloudBlob.getName(), cloudBlobList);
+                }
+            } else {
+                throw new RuntimeException("wrong blob type: " + blob.getUri());
+            }
+        }
+
+        private void getObjectSummaryIterator(Map<String, List<CloudBlob>> originalBlobMap) {
+            for (Map.Entry entry: originalBlobMap.entrySet()) {
+                long size = 0;
+                Map<String, CloudBlob> blobsWithSnapShotsMap = new TreeMap<>();
+                for (CloudBlob blob: (CloudBlob[]) entry.getValue()) {
+                    String key = "latest";
+                    if (blob.isSnapshot()) {
+
+                        key = blob.getSnapshotID();
+                    }
+                    blobsWithSnapShotsMap.put(key, blob);
+                    size += blob.getProperties().getLength();
+                }
+                ObjectSummary objectSummary = new ObjectSummary(entry.getKey().toString(), false, size);
+                objectSummary.setBlobsWithSnapShotsMap(blobsWithSnapShotsMap);
+                objectSummaries.add(objectSummary);
+            }
+        }
+    }
+
+    // TODO: implement directoryMode, using prefix+delimiter
+
+    @Override
+    public Iterable<ObjectSummary> children(ObjectSummary parent) {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public SyncObject loadObject(String identifier) throws ObjectNotFoundException {
+        return loadObject(identifier, config.isIncludeSnapShots());
+    }
+
+
+    private SyncObject loadObject(String identifier, boolean includeSnapShots) throws ObjectNotFoundException {
+        if (identifier == null) {
+            throw new ObjectNotFoundException(identifier);
+        }
+
+        if (includeSnapShots) {
+             List<BlobSyncObject> snapshots = loadSnapshots(identifier);
+             if (!snapshots.isEmpty()) {
+                 // use the latest version as source for return
+                 BlobSyncObject object = snapshots.get(snapshots.size() - 1);
+
+                 object.setProperty(PROP_BLOB_SNAPSHOTS, snapshots);
```

Comment:
I would set this as a member property since you're already implementing an Azure-specific object class.  Then when you add the versioned object abstraction, you can override the `getObjectVersions()` method (or whatever you end up calling it) and return this list.